src/boost_histogram/_internal/axis.py

```diff
@@ -155,30 +157,47 @@ def _repr_kwargs(self):
         """
 
         ret = ""
-        if self.options.growth:
+        if self.traits.growth:
             ret += ", growth=True"
-        elif self.options.circular:
+        elif self.traits.circular:
             ret += ", circular=True"
         else:
-            if not self.options.underflow:
+            if not self.traits.underflow:
                 ret += ", underflow=False"
-            if not self.options.overflow:
+            if not self.traits.overflow:
                 ret += ", overflow=False"
 
         return ret
 
     @property
     def options(self):
         """
+        DEPRECATED: use .trails instead.
         Return the options.  Fields:
           .underflow - True if axes captures values that are too small
           .overflow  - True if axes captures values that are too large
                        (or non-valid for category axes)
           .growth    - True if axis can grow
           .circular  - True if axis wraps around
         """
+        warnings.warn("DEPRECATED: use .traits instead", FutureWarning)
```

Comment:
You could steal and use the deprecated decorator from iminuit.

Reply:
I'll take a look, thanks!

Reply:
Fixed an issue present before the decorator with broken tab completion, and then added it. Made a few changes to the decorator, will quickly outline them in a comment box.